tests/subsys/audio_module/src/bad_param_test.c

```diff
@@ -0,0 +1,1171 @@
+		      -EINVAL, ret);
+}
+
+ZTEST(suite_audio_module_bad_param, test_connect_bad_type)
+{
+	int ret;
+
+	test_initialise_description(&test_description_tx, "Module Test 1",
+				    AUDIO_MODULE_TYPE_UNDEFINED, NULL);
+	test_initialise_description(&test_description_rx, "Module Test 2",
+				    AUDIO_MODULE_TYPE_UNDEFINED, NULL);
+
+	test_initialise_handle(&handle_tx, "TEST Connect 1", &test_description_tx,
+			       AUDIO_MODULE_STATE_CONFIGURED, NULL, NULL);
+	test_initialise_handle(&handle_rx, "TEST Connect 2", &test_description_rx,
+			       AUDIO_MODULE_STATE_CONFIGURED, NULL, NULL);
+
+	ret = audio_module_connect(&handle_tx, &handle_rx, false);
+	zassert_equal(ret, -ECANCELED, "Connect function did not return -ECANCELED (%d): ret %d",
+		      -ECANCELED, ret);
+
+	test_description_tx.type = AUDIO_MODULE_TYPE_IN_OUT;
+	test_description_rx.type = AUDIO_MODULE_TYPE_UNDEFINED;
+	ret = audio_module_connect(&handle_tx, &handle_rx, false);
+	zassert_equal(ret, -ECANCELED, "Connect function did not return -ECANCELED (%d): ret %d",
+		      -ECANCELED, ret);
+
+	handle_tx.state = AUDIO_MODULE_TYPE_UNDEFINED;
+	handle_rx.state = AUDIO_MODULE_TYPE_IN_OUT;
+	ret = audio_module_connect(&handle_tx, &handle_rx, false);
+	zassert_equal(ret, -ECANCELED, "Connect function did not return -ECANCELED (%d): ret %d",
+		      -ECANCELED, ret);
+
+	test_description_tx.type = AUDIO_MODULE_TYPE_INPUT;
+	test_description_rx.type = AUDIO_MODULE_TYPE_INPUT;
+	ret = audio_module_connect(&handle_tx, &handle_rx, false);
+	zassert_equal(ret, -ECANCELED, "Connect function did not return -ECANCELED (%d): ret %d",
+		      -ECANCELED, ret);
+
+	handle_tx.state = AUDIO_MODULE_TYPE_OUTPUT;
+	handle_rx.state = AUDIO_MODULE_TYPE_INPUT;
+	ret = audio_module_connect(&handle_tx, &handle_rx, false);
+	zassert_equal(ret, -ECANCELED, "Connect function did not return -ECANCELED (%d): ret %d",
+		      -ECANCELED, ret);
+
+	handle_tx.state = AUDIO_MODULE_TYPE_IN_OUT;
+	handle_rx.state = AUDIO_MODULE_TYPE_OUTPUT;
+	ret = audio_module_connect(&handle_tx, &handle_rx, false);
+	zassert_equal(ret, -ECANCELED, "Connect function did not return -ECANCELED (%d): ret %d",
+		      -ECANCELED, ret);
+}
+
+ZTEST(suite_audio_module_bad_param, test_connect_bad_state)
+{
+	int ret;
+
+	test_initialise_description(&test_description_tx, "Module Test 1", AUDIO_MODULE_TYPE_IN_OUT,
+				    NULL);
+	test_initialise_description(&test_description_rx, "Module Test 2", AUDIO_MODULE_TYPE_IN_OUT,
+				    NULL);
+
+	test_initialise_handle(&handle_tx, "TEST Connect 1", &test_description_tx,
+			       AUDIO_MODULE_STATE_UNDEFINED, NULL, NULL);
+	test_initialise_handle(&handle_rx, "TEST Connect 2", &test_description_rx,
+			       AUDIO_MODULE_STATE_UNDEFINED, NULL, NULL);
+
+	ret = audio_module_connect(&handle_tx, &handle_rx, false);
+	zassert_equal(ret, -ECANCELED, "Connect function did not return -ECANCELED (-129): ret %d",
```

Comment:
I think all numbers (-129) should be removed. This may change and then we have to update the prints.

Reply:
This should have been a %d in these few places to be consistent with the other tests